### PR TITLE
Add a tool to support automatically maintaining sort order for sorted sections.

### PR DESCRIPTION
**Pull request by nuclearsandwich**
I started down this path because I was separately trying to write a script that would insert values in the correct places and decided that it would be easier to write one script to add things wherever and then another to sort the entire section.

As far as I can tell, every sort change in the tf files is one that my own manual editing also performs and these have just accumulated due to the lack of tools like this one to catch them.

While drafting #476, which this replaces, I realized that with a bit more cleverness (always a risk) I could do away with the explicit sigils and this is the result.

### CODE REVIEW

File: bin/sort-fields [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+function tmpfile_name(filename) {
+  return "." filename ".tmp"
+}
+
+function panic(msg) {
+	print FILENAME ":" FNR ": " msg > "/dev/stderr"
+	exit(1)
+}
+
```
Comment (cottsay) [inline bin/sort-fields:12]: Does the extra newline have significance here?

File: bin/sort-fields [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	}
+	if (!end_pattern) {
+		end_pattern = "keep-sorted end"
+	}
+
+	files_n = 0
+	delete origfiles
+	delete tmpfiles
+
+
```
Comment (cottsay) [inline bin/sort-fields]: Same with these. Significant?

File: bin/sort-tf-fields [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+			start_pattern='^  [_a-z0-9]+_(admins|team|repositories) = \\[$' 
+			end_pattern='^  \\]$'
+	esac
+	bin/sort-fields \
+		-v start_pattern="$start_pattern" \
+		-v end_pattern="$end_pattern" \
+		"$tf_file"
+done
+
+
```
Comment (cottsay) [inline bin/sort-tf-fields]:
```suggestion
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -5,6 +5,14 @@ Changes to the organization structure are managed by the GitHub Terraform provid
 Each set of repositories is maintained by a release team with a name that matches the repository name (such as `octomap`), or in some cases, the umbrella project which maintains multiple release repositories (such as `navigation` or `tooling_wg`).
 Each release team's configuration is stored in its own file: `RELEASE_TEAM.tf` so, for example, configuration for the `tooling_wg` team is stored in `tooling_wg.tf`.
 
+## Formatting
+
+`terraform fmt` can be used to enforce consistent formatting of files in this project.
+Several files include long lists of local variables, repository names, or member usernames.
+There is a script this project which will automatically sort these lines which you can use to check and enforce sort order before committing:
```
Comment (cottsay) [inline CONTRIBUTING.md]:
```suggestion
There is a script in this project which will automatically sort these lines which you can use to check and enforce sort order before committing:
```

**Review by cottsay — CHANGES_REQUESTED** [review]
Some whitespace inconsistencies to address, but nothing big.

I'm not an awk nerd, but I don't see anything alarming.

It would be nice if you could invoke this as part of the CI action.

Does the script return non-zero if changes were made?

File: bin/sort-fields (hunk 1 above)
Comment (nuclearsandwich) [reply bin/sort-fields:12]: Nah all this whitespace is just the result of me typing faster than I'm reading. I'll clean it all up.

**Comment by nuclearsandwich** [comment]
> It would be nice if you could invoke this as part of the CI action.

This is planned for another Friday afternoon.
 
> Does the script return non-zero if changes were made?

It does not. The approach I'd take to CI-ify this would be to run the script and then raise an error if the git workspace is dirty.

**Comment by nuclearsandwich** [comment]
The reason I'm rebasing fixups is that I'd like to Rebase-merge / fast-forward these commits rather than squashing them so that the content commits are separate from the formatting ones.

**Review by cottsay — APPROVED** [review]

**Comment by nuclearsandwich** [comment]
Of all the... I _just_ said that I wanted to rebase / fast-forward these changes and here I go clicking the squash button the moment I see it lit up. Well I have push powers on latest and just you watch me commit more acts of skullduggery to clean up after my mistakes.

**Comment by nuclearsandwich** [comment]
`latest` has been pushed to a356263, achieving the fast-forward. I will answer for my crimes someday.

File: bin/sort-tf-fields [hunk 5]
```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+# vim:ts=2:sw=2:noet
+
+for tf_file in "$@"; do
+	case "$tf_file" in
+		00-repositories.tf)
+			start_pattern='^  organization_repostiories = setunion\\($'
```
Comment (nuclearsandwich) [inline bin/sort-tf-fields:7]: STEVEN!

Comment (cottsay) [reply bin/sort-tf-fields:7]: > start_pattern='^  organization_repostiories = setunion\\($'

> organization_repostiories

> **repostiories**

> ### **repostiories**

> # **repostiories**

**Pull request merged**